src/main/resources/site/pages/default/default.js

```diff
@@ -1,77 +1,112 @@
-        stk.data.deleteEmptyProperties(siteConfig);
+function buildBodyClass(site, siteConfig, content, params) {
+    var bodyClass = '';
+    if (siteConfig.backgroundImage) {
+        bodyClass += 'custom-background ';
+    }
+    if ((content._path == site._path) && stk.data.isEmpty(params)) {
+        bodyClass += 'home blog ';
+    }
+    if (params.cat || params.tag || params.author) {
+        bodyClass += ' archive ';
+    }
+    if (content.type == app.name + ':post' || content.type == 'portal:page-template') {
+        bodyClass += 'single single-post single-format-standard ';
+    }
+    if (params.author) {
+        bodyClass += 'author '
+    }
 
-        var content = portal.getContent();
+    return bodyClass;
+}
 
-        var bodyClass = '';
-        var backgroundImage;
-        if (siteConfig.backgroundImage) {
-            var bgImageUrl = portal.imageUrl({
-                id: siteConfig.backgroundImage,
-                scale: '(1,1)',
-                format: 'jpeg'
-            });
 
-            backgroundImage = '<style type="text/css" id="custom-background-css">body.custom-background { background-image: url("' +
-                bgImageUrl + '"); background-repeat: repeat; background-position: top left; background-attachment: scroll; }</style>';
+function getPageItemClass(targetPath, currentContentPath) {
+    return targetPath == currentContentPath ? 'current_page_item' : 'page_item';
+}
 
-            bodyClass += 'custom-background ';
-        }
-        if ((content._path == site._path) && stk.data.isEmpty(up)) {
-            bodyClass += 'home blog ';
-        }
-        if (up.cat || up.tag || up.author) {
-            bodyClass += ' archive ';
-        }
-        if (content.type == app.name + ':post' || content.type == 'portal:page-template') {
-            bodyClass += 'single single-post single-format-standard ';
-        }
-        if (up.author) {
-            bodyClass += 'author '
-        }
 
-        var footerText = siteConfig.footerText ? portal.processHtml({value: siteConfig.footerText}): 'Configure footer text.';
-
-        /*'<a href="https://enonic.com/" title="Fastest way from idea to ' +
-            'digital experience." rel="generator">Proudly powered by Enonic XP</a> <span class="sep"> | </span> Theme: Superhero by ' +
-            '<a href="https://wordpress.com/themes/" rel="designer">WordPress.com</a>.';*/
-
-        var isFragment = content.type === 'portal:fragment';
-        var model = {
-            site: site,
-            bodyClass: bodyClass,
-            backgroundImage: backgroundImage,
-            mainRegion: isFragment ? null : content.page.regions['main'],
-            isFragment: isFragment,
-            content: content,
-            menuItems: menuItems,
-            footerText: footerText,
-            headerStyle: req.mode == 'edit' ? 'position: absolute;' : null
-        };
+function insertPageurlsAndClassesIntoMenuItems(menuItems, contentPath) {
```

Comment:
~~pen...~~ ~~password~~ Method name is too long. 
How about setupMenuStructure? 

Do we want return the menu items from the method? Since most methods don't modify passed params?

Reply:
Generating a new returned deep array without mutating the old one is real easy is ES6. In ecma-nashorn, not so much. Or rather, I have to figure out the easiest way that's supported.

Just mutating it and being **very** clear about that in the method name, is so much easier. Seemed a fair-enough way out. Low hanging fruit and all that.

Reply:
I was just thinking `let menutitems = setupMenuStructure(menuItems);`
But I'm okay with mutating passed arguments, just not that long method name.